Use vaadin themable.

manolo opened the pull request:
Fixes #65

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-split-layout/74)
<!-- Reviewable:end -->

Saulis commented:
Looks ok in general, but the visual screen test results seem to look different from before?

---

Review status: 0 of 16 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-split-layout.html, line 49 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-Kn5d_JlkzaqChZH1YhU:-Kn5d_JlkzaqChZH1YhV:b-5sq694) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2598c2f58a8450698b9ffa2417da7165b9d0c3fd/vaadin-split-layout.html#L49)):*
> ```HTML
>       }
> 
>       :host(:not([vertical])) > [part=splitter] {
> ```

nitpick: missing double-quotes around the attribute value: `[part="splitter"]`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-Kn5eKzm3-pSOFPNQRo0:b-j296tr)*
<!-- Sent from Reviewable.io -->

manolo commented:
For some reason, font icons produce blear borders. I have opened the fonts in a web font editor and border seems ok though. Probably someone else more familiar on this sort of tools could take a look. 

---

Review status: 0 of 16 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-split-layout.html, line 49 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-Kn5d_JlkzaqChZH1YhU:-Kn88dtQtjomtI2IIRJq:b-896fix) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2598c2f58a8450698b9ffa2417da7165b9d0c3fd/vaadin-split-layout.html#L49)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

nitpick: missing double-quotes around the attribute value: `[part="splitter"]`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-Kn84cl9OSnvk3yrWFS5:b-pxmt6n)*
<!-- Sent from Reviewable.io -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/vaadin-split-layout?pullRequest=74) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/vaadin-split-layout?pullRequest=74) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: manolo<br/>:x: Saulis<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/vaadin-split-layout?pullRequest=74) it.</sub>

Saulis commented:
Ahem. Looks like this PR got closed after I deleted the branch after merging #73. Sorry :-)

Saulis commented:
Restored the branch, please rebase this PR on top of `master`

manolo commented:
Done

---

Review status: 0 of 16 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-KndCxXhdvB3xdixIARK:bb6un1j)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Review status: 0 of 16 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-split-layout.html, line 15 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KnhihkArGsAkyqugZm5:-KnhihkBfiVGtS3WAscP:b-lvvvl2) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2d652a88b78d37c06cdeb1697043c98b994e373e/vaadin-split-layout.html#L15)):*
> ```HTML
>   @font-face {
>     font-family: 'vaadin-split-layout-icons';
>     src: url(data:application/font-woff;base64,d09GRgABAAAAAAcsAAoAAAAABuQAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABPUy8yAAAA9AAAAGAAAABg+BMGCGNtYXAAAAFUAAABQgAAAUIADeqpZ2x5ZgAAApgAAAFYAAABWJhloHpoZWFkAAAD8AAAADYAAAA2DP6ajGhoZWEAAAQoAAAAJAAAACQHMAQyaG10eAAABEwAAAAUAAAAFA9tBUZsb2NhAAAEYAAAAAwAAAAMAHIAyG1heHAAAARsAAAAIAAAACAACQAybmFtZQAABIwAAAJSAAACUgRElDpwb3N0AAAG4AAAAEwAAABMNMd8dwAEA9sBkAAFAAACmQLMAAAAjwKZAswAAAHrADMBCQAAAAAAAAAAAAAAAAAAAAEQAAAAAAAAAAAAAAAAAAAAAEDpAOkDA8D/wABAA8AAQAAAAAEAAAAAAAAAAAAAACAAAAAAAAMAAAADAAAAHAABAAAAAAA8AAMAAQAAABwABAAgAAAABAAEAAEAAOkD//8AAOkA//8XAQABAAAAAAAAAQYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAeoAdQIWAwsAAwAAATMRIwHqLCwDC/1qAAAAAQBsAaYDAgHbAAMAABM1IRVsApYBpjU1AAAAAAMB0gDeAi0CogAPAB8ALwAAAA4CFB4CMj4CNC4DDgIUHgIyPgI0LgICDgIUHgIyPgI0LgIB9hEMBwcMERMQDAgIDBATEQwHBwwRExAMCAgMEBMRDAcHDBETEAwICAwQAe8IDBATEQwHBwwRExAMCLMHDRATEA0HBw0QExANB/6WBwwREhEMBwcMERIRDAcAAAMBHgGTAuIB7QAPAB8ALwAAAB4CMj4CNC4CIg4DHgIyPgI0LgIiDgIEHgIyPgI0LgIiDgIB0QcNEBMQDQcHDRATEA0HswcMERMQDAgIDBATEQwHAWoHDBATEQwHBwwRExAMBwG3EQwHBwwREhEMBwcMERIRDAcHDBESEQwHBwwREhEMBwcMERIRDAcHDBEAAAEAAAABAADKRfaZXw889QALBAAAAAAA1VRE+AAAAADVHhEXAGwAdQMCAwsAAAAIAAIAAAAAAAAAAQAAA8D/wAAABAAAbABsAwIAAQAAAAAAAAAAAAAAAAAAAAUAAAAABAAB6gNuAGwD/wHSBAABHgAAAAAADgAcAGQArAABAAAABQAwAAMAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAADgCuAAEAAAAAAAEAGAAAAAEAAAAAAAIABwAYAAEAAAAAAAMAGAAfAAEAAAAAAAQAGAA3AAEAAAAAAAUACwBPAAEAAAAAAAYAGABaAAEAAAAAAAoAGgByAAMAAQQJAAEAMACMAAMAAQQJAAIADgC8AAMAAQQJAAMAMADKAAMAAQQJAAQAMAD6AAMAAQQJAAUAFgEqAAMAAQQJAAYAMAFAAAMAAQQJAAoANAFwdmFhZGluLWRhdGUtcGlja2VyLWljb25zUmVndWxhcnZhYWRpbi1kYXRlLXBpY2tlci1pY29uc3ZhYWRpbi1kYXRlLXBpY2tlci1pY29uc1ZlcnNpb24gMS4wdmFhZGluLWRhdGUtcGlja2VyLWljb25zRm9udCBnZW5lcmF0ZWQgYnkgSWNvTW9vbi4AdgBhAGEAZABpAG4ALQBkAGEAdABlAC0AcABpAGMAawBlAHIALQBpAGMAbwBuAHMAUgBlAGcAdQBsAGEAcgB2AGEAYQBkAGkAbgAtAGQAYQB0AGUALQBwAGkAYwBrAGUAcgAtAGkAYwBvAG4AcwB2AGEAYQBkAGkAbgAtAGQAYQB0AGUALQBwAGkAYwBrAGUAcgAtAGkAYwBvAG4AcwBWAGUAcgBzAGkAbwBuACAAMQAuADAAdgBhAGEAZABpAG4ALQBkAGEAdABlAC0AcABpAGMAawBlAHIALQBpAGMAbwBuAHMARgBvAG4AdAAgAGcAZQBuAGUAcgBhAHQAZQBkACAAYgB5ACAASQBjAG8ATQBvAG8AbgAuAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABQAFAAABAgEDAQQBBQd1bmlFOTAwB3VuaUU5MDEHdW5pRTkwMgd1bmlFOTAz) format('woff');
> ```

Now that the "handle" is a themable part, we could just make it a 40x40 flexbox in the default styles and center a normal sized icon in it. I think the icons specified here have lots of empty space around them which dictates the handle bounds.

---

*[vaadin-split-layout.html, line 81 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KnhlGrpbap2fPVv_00s:-KnhlGrpbap2fPVv_00t:b-vndzdy) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2d652a88b78d37c06cdeb1697043c98b994e373e/vaadin-split-layout.html#L81)):*
> ```HTML
>     <slot id="primary" name="primary"></slot>
>     <div part="splitter" id="splitter" on-track="_onHandleTrack" on-down="_preventDefault">
>       <slot name="handle">
> ```

Is this feature still relevant after the new theming approach?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 16 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-split-layout.html, line 15 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KnhihkArGsAkyqugZm5:-Knhnxb5JSoCL1gZ38VH:b-3ka6i0) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2d652a88b78d37c06cdeb1697043c98b994e373e/vaadin-split-layout.html#L15)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Now that the "handle" is a themable part, we could just make it a 40x40 flexbox in the default styles and center a normal sized icon in it. I think the icons specified here have lots of empty space around them which dictates the handle bounds.
</blockquote></details>

I don't follow. You mean modify the font ?

---

*[vaadin-split-layout.html, line 81 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KnhlGrpbap2fPVv_00s:-KnhnUh7z7j6Nkt-Hazz:bklpsc8) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2d652a88b78d37c06cdeb1697043c98b994e373e/vaadin-split-layout.html#L81)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Is this feature still relevant after the new theming approach?
</blockquote></details>

I guess the user could still want to replace the handle with any html content like a paper-icon etc. 
There might be cases where the font content is not enough

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-split-layout.html, line 15 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KnhihkArGsAkyqugZm5:-Knhogp10320UOntcVy1:bctdoxw) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2d652a88b78d37c06cdeb1697043c98b994e373e/vaadin-split-layout.html#L15)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I don't follow. You mean modify the font ?
</blockquote></details>

Yeah, IIRC the space around the handle icon was due to it being an SVG earlier.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-split-layout.html, line 81 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KnhlGrpbap2fPVv_00s:-KnhpnnCizkbxOiGiPuU:b-ykj17a) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2d652a88b78d37c06cdeb1697043c98b994e373e/vaadin-split-layout.html#L81)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I guess the user could still want to replace the handle with any html content like a paper-icon etc. 
There might be cases where the font content is not enough
</blockquote></details>

Of course, there can always be edge cases. But the point is that would the new support for modifying the icons with theme be enough for 99.9% of the cases. Has this slot now become one of those features that we support but virtually no-one actually needs anymore?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 16 files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

*[vaadin-split-layout.html, line 15 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KnhihkArGsAkyqugZm5:-Knn_R72sRyeOvyq2GzP:b-rnv3yr) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2d652a88b78d37c06cdeb1697043c98b994e373e/vaadin-split-layout.html#L15)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Yeah, IIRC the space around the handle icon was due to it being an SVG earlier.
</blockquote></details>

Now the browser font renderer is giving the space that the font needs with the given font-size.
Since we are using lines here differences exist, with other icons like ellipsis, etc differences are not so noticeable.
I think this issues can be fixed by the user selecting another font. Also we could work more trying to understand how fonts work and use better font editors, but I think is out of this PR scope.

---

*[vaadin-split-layout.html, line 81 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KnhlGrpbap2fPVv_00s:-Knn_CMIt8qS1rniPW0l:b8eh43v) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2d652a88b78d37c06cdeb1697043c98b994e373e/vaadin-split-layout.html#L81)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Of course, there can always be edge cases. But the point is that would the new support for modifying the icons with theme be enough for 99.9% of the cases. Has this slot now become one of those features that we support but virtually no-one actually needs anymore?
</blockquote></details>

Removed support for slot. Also removed custom demos, since all of remaining were based on slots.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 16 files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

*[vaadin-split-layout.html, line 15 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KnhihkArGsAkyqugZm5:-Knpxq5K7CzBXCFQ7dws:b-up5wzz) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/2d652a88b78d37c06cdeb1697043c98b994e373e/vaadin-split-layout.html#L15)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Now the browser font renderer is giving the space that the font needs with the given font-size.
Since we are using lines here differences exist, with other icons like ellipsis, etc differences are not so noticeable.
I think this issues can be fixed by the user selecting another font. Also we could work more trying to understand how fonts work and use better font editors, but I think is out of this PR scope.
</blockquote></details>

Removed fonts. Using a styled element instead.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Reviewed 1 of 11 files at r3.
Review status: 1 of 16 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-split-layout.html, line 235 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-Ko7AYEWNaOOXrysHXgA:-Ko7AYEWNaOOXrysHXgB:b4i1nhq) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/8a155ad945eee6deae331b33d17efc437ca5e112/vaadin-split-layout.html#L235)):*
> ```HTML
>        *
>        * You can replace the default splitter handle with a custom handle element
>        * by using the `handle` slot.
> ```

It seems that support for slotted handle was removed later in https://github.com/vaadin/vaadin-split-layout/commit/9da6acad1e14af859cdd49956b4676c6508149a8 . (the same for html block below)

---

*[test/visual/customized.html, line 18 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-Ko7BGgd1h5X68KBQnM2:-Ko7BGgd1h5X68KBQnM3:b-eqi8y9) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/8a155ad945eee6deae331b33d17efc437ca5e112/test/visual/customized.html#L18)):*
> ```HTML
>         }
> 
>         [part=handle] {
> ```

nitpick: missing double-quotes around the attribute value: [part="handle"] and [part="splitter"]

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 1 of 16 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-split-layout.html, line 235 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-Ko7AYEWNaOOXrysHXgA:-Ko7TixxZ12bIuUk048c:b-896fix) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/8a155ad945eee6deae331b33d17efc437ca5e112/vaadin-split-layout.html#L235)):*
<details><summary><i>Previously, YuriyVaadin (Yuriy Yevstihnyeyev) wrote…</i></summary><blockquote>

It seems that support for slotted handle was removed later in https://github.com/vaadin/vaadin-split-layout/commit/9da6acad1e14af859cdd49956b4676c6508149a8 . (the same for html block below)
</blockquote></details>

Done.

---

*[test/visual/customized.html, line 18 at r3](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-Ko7BGgd1h5X68KBQnM2:-Ko7TuygoFMf_f2awhhp:b-896fix) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/8a155ad945eee6deae331b33d17efc437ca5e112/test/visual/customized.html#L18)):*
<details><summary><i>Previously, YuriyVaadin (Yuriy Yevstihnyeyev) wrote…</i></summary><blockquote>

nitpick: missing double-quotes around the attribute value: [part="handle"] and [part="splitter"]
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

limonte commented:
I have the same question as Sauli had:

> the visual screen test results seem to look different from before?

Looks like `test/visual/customized.html` doesn't work, yellow handles are expected.

---

Review status: 0 of 18 files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-KoGr9N_CBHtuwNWotu2:b-dpy4n3)*
<!-- Sent from Reviewable.io -->

manolo commented:
> Looks like test/visual/customized.html doesn't work, yellow handles are expected.

Yep, theming requires more properties to make it look like before. Done.

---

Review status: 0 of 18 files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-KoGvD278v59UOWf6T12:b-6mgrw7)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 2 of 17 files at r1, 7 of 11 files at r3, 2 of 3 files at r4, 4 of 4 files at r5, 4 of 4 files at r6.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[vaadin-split-layout.html, line 15 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHN2Zt_6j2-U_iH0fh:-KoHN2Zt_6j2-U_iH0fi:b3y6mkv) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/vaadin-split-layout.html#L15)):*
> ```HTML
>   <template>
>     <style>
>       [part="handle"] {
> ```

The handle size should be part of the core style. This is a rather functional style, it doesn’t affect much except the touch target area. Every theme should have a large handle touch area anyway.

---

*[vaadin-split-layout.html, line 22 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHSbFbhID3hT2UJPof:-KoHSbFbhID3hT2UJPog:b-hzhve3) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/vaadin-split-layout.html#L22)):*
> ```HTML
>       [part="handle"]::after {
>         content: "";
>         position: absolute;
> ```

I think, centering with 50% position + negative transform is inappropriate here.

Would do instead:
```css
      [part="handle"]::after {
        content: "";
        display: block;
        margin: 8px auto;
        width: 2px;
        height: 24px;
        background: white;
      }
```

Another option with more lines but less magic numbers:
```css
      [part="handle"]::after {
        content: "";
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        margin: auto;
        width: 2px;
        height: 24px;
        background: white;
      }
```

---

*[vaadin-split-layout.html, line 62 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHNLAMLABBT4__zDRV:-KoHNLAMLABBT4__zDRW:b-ap7zkt) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/vaadin-split-layout.html#L62)):*
> ```HTML
>         min-width: 8px;
>         min-height: 8px;
>         background: #ccc;
> ```

The color here is theme-opinionated, should be in the default theme.

Maybe the size should be in the default theme too.

---

*[demo/theming.html, line 26 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHP4KTC4XxVF4CpS2M:-KoHP4KTC4XxVF4CpS2N:b89ofuj) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/demo/theming.html#L26)):*
> ```HTML
>           <template>
>             <style>
>               [part="handle"]::after {
> ```

Theme demo should use `~=` part selectors, I suppose.

---

*[test/visual/customized.html, line 10 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHVWJBr_4-u2kv4MgE:-KoHVWJC7HmJ4h30FKaI:b-442p9c) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/test/visual/customized.html#L10)):*
> ```HTML
> 
> <body>
>   <dom-module id="vaadin-split-layout-style" theme-for="vaadin-split-layout">
> ```

Should we reuse theming demo in this test?

Personally, I try to reuse demo code in the tests as much as I can:
- More straightforward to maintain, just copy-paste to test when the demo is updated. Less thinking involved, compared to having different demos and tests.
- Makes the demos somewhat tested too.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

limonte commented:
I miss nice examples from the previous version:

https://cdn.vaadin.com/vaadin-core-elements/master/vaadin-split-layout/demo/custom.html#customizing-vertical-and-horizontal-layouts-separately
https://cdn.vaadin.com/vaadin-core-elements/master/vaadin-split-layout/demo/custom.html#customizing-nested-layouts
https://cdn.vaadin.com/vaadin-core-elements/master/vaadin-split-layout/demo/custom.html#thin-lines
https://cdn.vaadin.com/vaadin-core-elements/master/vaadin-split-layout/demo/custom.html#custom-handle

I would like to the the same result achieved by vaadin-themable.

---

Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-KoHlj3IAdpgRXpuFiyd:bt17in7)*
<!-- Sent from Reviewable.io -->

manolo commented:
Those are not possible right now because 1.- we removed support for slots and 2.- using themable they would be very verbose and we would need a different page per theme. In case someone asks for them we might do in jsfiddle.
We already agreed to remove them.

---

Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[vaadin-split-layout.html, line 62 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHNLAMLABBT4__zDRV:-KoHppgPPllDBFlT20_I:bu2uqy3) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/vaadin-split-layout.html#L62)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

The color here is theme-opinionated, should be in the default theme.

Maybe the size should be in the default theme too.
</blockquote></details>

I think color and size are needed for the element work if default theme is removed, this is the reason they are here. And anyways it is easily overridable in custom themes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-KoHp-6Gp7QVqbubn8FV:b-3g42m9)*
<!-- Sent from Reviewable.io -->

platosha commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[vaadin-split-layout.html, line 62 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHNLAMLABBT4__zDRV:-KoHqKfzdPmqAS5FC85d:b-eh6vp1) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/vaadin-split-layout.html#L62)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I think color and size are needed for the element work if default theme is removed, this is the reason they are here. And anyways it is easily overridable in custom themes.
</blockquote></details>

Should defenetely work without the color, probably even size.

`8px` was a material design based choice here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[vaadin-split-layout.html, line 22 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHSbFbhID3hT2UJPof:-KoJ0dAWnz8ZBJrG6ycz:b-p8znj2) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/vaadin-split-layout.html#L22)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

I think, centering with 50% position + negative transform is inappropriate here.

Would do instead:
```css
      [part="handle"]::after {
        content: "";
        display: block;
        margin: 8px auto;
        width: 2px;
        height: 24px;
        background: white;
      }
```

Another option with more lines but less magic numbers:
```css
      [part="handle"]::after {
        content: "";
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        margin: auto;
        width: 2px;
        height: 24px;
        background: white;
      }
```
</blockquote></details>

Why is inappropriate now, and it was ok in the previous version ?

---

*[vaadin-split-layout.html, line 62 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHNLAMLABBT4__zDRV:-KoIycyWLKu5G9gLob0L:b-41s5p1) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/vaadin-split-layout.html#L62)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Should defenetely work without the color, probably even size.

`8px` was a material design based choice here.
</blockquote></details>

`Default styles should contain functional styles, which are required to make the element work (overall layout, showing and hiding parts etc.)`

`Purely visual styles such as box-shadows, color, etc. are okay to add when it’s expected that most themes will either add or override those in any case. Just make sure they are easy to override.`

In this case the splitter without any size and color makes the element un-usable. They are expected to override for almost themes, and they are easy to override. I don't see any better reason for removing them.

I would change the values, if you are against having specific material values, probably to

```
min-width: 2px;
min-height: 2px;
background: black;
```

---

*[demo/theming.html, line 26 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHP4KTC4XxVF4CpS2M:-KoJ1FBF49dEomLHBR3I:b-5y4lxe) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/demo/theming.html#L26)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Theme demo should use `~=` part selectors, I suppose.
</blockquote></details>

well it's demo code, we are not writing generic themes here.

---

*[test/visual/customized.html, line 10 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHVWJBr_4-u2kv4MgE:-KoJ0JDgbPmuvg40M2p7:b-c1u4mw) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/test/visual/customized.html#L10)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Should we reuse theming demo in this test?

Personally, I try to reuse demo code in the tests as much as I can:
- More straightforward to maintain, just copy-paste to test when the demo is updated. Less thinking involved, compared to having different demos and tests.
- Makes the demos somewhat tested too.
</blockquote></details>

I would that in a separate PR. For this PR I want to show what are the visual differences before and after the modifications

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 17 of 18 files reviewed at latest revision, 7 unresolved discussions.

---

*[vaadin-split-layout.html, line 15 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoHN2Zt_6j2-U_iH0fh:-KoJ4hx-484QiSVINys0:bmaigow) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/36098df93333c9c7e06bd954899046cb786f495e/vaadin-split-layout.html#L15)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

The handle size should be part of the core style. This is a rather functional style, it doesn’t affect much except the touch target area. Every theme should have a large handle touch area anyway.
</blockquote></details>

Not having the handler size, the component works, so I think it's not a fundamental functional feature, this is the reason it's in the theme.

I don't have a strong preference so moving it to the default styles.

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

limonte commented:
I changed a little bit theming demo page to be a little bit prettier and cross-platform.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Review status: 16 of 20 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-KoHuhA4T051FSLmA6D8:b-md2z1i)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 8 of 20 files reviewed at latest revision, 4 unresolved discussions.

---

*[demo/theming.html, line 42 at r8](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoM5XdClRhmB60oX4sp:-KoM5XdClRhmB60oX4sq:b-kl5ilp) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/1f8780d441380c2e06e14f266b51da17ea6bd651/demo/theming.html#L42)):*
> ```HTML
>               }
> 
>               :host([vertical]) [part~="handle"]::after {
> ```

Nit: the handler seems a bit misplaced (not in the middle)

![Screen Shot 2017-07-06 at 10.51.57.png](https://files.reviewableusercontent.io/images/reviewable/161853/w0VJbUqzhbJYrYOEiwHmUw8c/Screen%20Shot%202017-07-06%20at%2010.51.57.png)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

Toshegg commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 8 of 20 files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-KoMQ7Ojj3sjN3_eIePL:bnfp4nl)*
<!-- Sent from Reviewable.io -->

limonte commented:
Review status: 8 of 20 files reviewed at latest revision, 4 unresolved discussions.

---

*[demo/theming.html, line 42 at r8](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoM5XdClRhmB60oX4sp:-KoMlomHaQts-zPsit3O:b-qifkf1) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/1f8780d441380c2e06e14f266b51da17ea6bd651/demo/theming.html#L42)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Nit: the handler seems a bit misplaced (not in the middle)

![Screen Shot 2017-07-06 at 10.51.57.png](https://files.reviewableusercontent.io/images/reviewable/161853/w0VJbUqzhbJYrYOEiwHmUw8c/Screen%20Shot%202017-07-06%20at%2010.51.57.png)

</blockquote></details>

It's impossible to get pixel-perfect cross-browser result with UTF symbols. If we want pixel-perfect perfection in demos (do we?), images should be used. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 8 of 20 files reviewed at latest revision, 4 unresolved discussions.

---

*[demo/theming.html, line 42 at r8](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-KoM5XdClRhmB60oX4sp:-KoQY85PdcEu90HPB2E3:b-fvys5n) ([raw file](https://github.com/vaadin/vaadin-split-layout/blob/1f8780d441380c2e06e14f266b51da17ea6bd651/demo/theming.html#L42)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

It's impossible to get pixel-perfect cross-browser result with UTF symbols. If we want pixel-perfect perfection in demos (do we?), images should be used. 
</blockquote></details>

I think, if you increase the splitter width misplacement is less noticeable

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->

manolo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 17 files at r1, 2 of 11 files at r3, 1 of 3 files at r4, 4 of 4 files at r5, 11 of 12 files at r8, 1 of 1 files at r9.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74#-:-KogjC5o08_PNZNZMbl-:bnfp4nl)*
<!-- Sent from Reviewable.io -->

limonte commented:
Reviewed 2 of 17 files at r1, 2 of 11 files at r3, 1 of 3 files at r4, 4 of 4 files at r5, 11 of 12 files at r8, 1 of 1 files at r9.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-split-layout/74)*
<!-- Sent from Reviewable.io -->
